packages/astro/src/internal/__astro_component.ts

```diff
@@ -187,7 +187,8 @@ export function __astro_component(Component: any, metadata: AstroComponentMetada
     }
 
     // If we ARE hydrating this component, let's generate the hydration script
-    const astroId = hash.unique(html);
+    const stringifiedProps = JSON.stringify(props);
+    const astroId = hash.unique(html + stringifiedProps);
```

Comment:
what happens if two components with the same props? It sounds like we end up with two `astro-root` components with the same `uid` attribute. 

Curious if this is something that we should optimize for (just render all matching components with the same hydrate script) or be super explicit about (each component gets a completely unique id, not based on html).

Reply:
We should get rid of these inline scripts altogether imo and colocate everything in the head. A small inlined custom element would make these a lot easier.

But I think we should merge this fix in first because this seems like a bad bug to me.

Reply:
> what happens if two components with the same props? It sounds like we end up with two `astro-root` components with the same `uid` attribute. 
> 
> 
> 
> Curious if this is something that we should optimize for (just render all matching components with the same hydrate script) or be super explicit about (each component gets a completely unique id, not based on html).

My opinion is that the way it currently works goes against established patterns of hydration, and it's not the expected behavior.

Using a modified version of the React counter component as an example:

```jsx

import React, { useState } from 'react';

export default function Counter({ initialValue, children }) {
  const [count, setCount] = useState(initialValue);
  const add = () => setCount((i) => i + 1);
  const subtract = () => setCount((i) => i - 1);

  return (
    <>
      <div className="counter">
        <button onClick={subtract}>-</button>
        <pre>{count}</pre>
        <button onClick={add}>+</button>
      </div>
      <div className="children">{children}</div>
    </>
  );
}

```

And then use two instances with different initial values in Astro:

```jsx
<Counter initialValue={0} />
<Counter initialValue={10} />
```

The above code would result in both Counter components being hydrated with the same initialValue.

I believe that using the components props as part of the hash generation gives us both the current optimization (identical components with identical props will get the same hydrate script) _and_ correctly hydrates components that differ only in what prop values get set from Astro.